ui/address/details/AddressBalance.tsx

```diff
@@ -65,6 +77,67 @@ const AddressBalance = ({ data, isLoading }: Props) => {
     handler: handleNewCoinBalanceMessage,
   });
 
+  const value = data.coin_balance || '0';
+  const exchangeRate = data.exchange_rate;
+  const decimals = String(config.chain.currency.decimals);
+  const accuracyUsd = 2;
+  const accuracy = 8;
+
+  const onGetGasClick = React.useCallback(() => {
+    mixpanel.logEvent(mixpanel.EventTypes.BUTTON_CLICK, { Content: 'Get gas', Source: 'address' });
+  }, []);
+
+  let getGasButton = null;
+  const { usd: usdResult } = getCurrencyValue({ value, accuracy, accuracyUsd, exchangeRate, decimals });
+
+  if (getGasFeature.isEnabled && !data?.is_contract && Number(usdResult) < getGasFeature.usdThreshold) {
```

Comment:
here usdResult can be undefined, so you're comparing NaN with a number

do we want to show the button on empty addresses (like https://eth.blockscout.com/address/0xCBE27aeac63D13BE49625166AAf1BCF4F8e4bE01)? 
in any case, add a type check here

Reply:
> here usdResult can be undefined, so you're comparing NaN with a number

What's wrong with the condition in this case? It will return false as desired

Reply:
it's not obvious that false is expected here and is not a bug